models/DataObject/ClassDefinition/Data.php

```diff
@@ -1308,4 +1291,14 @@ public function isForbiddenName(): bool
     {
         return in_array($this->getName(), self::FORBIDDEN_NAMES);
     }
+
+    public function jsonSerialize(): mixed
+    {
+        $tempObj = clone $this; // we clone, so that we only get public properties of the object
+        $data = get_object_vars($tempObj);
+        $data['fieldtype'] = $this->getFieldType();
+        $data['datatype'] = 'data';
+        unset($data['blockedVarsForExport']);
+        return $data;
+    }
```

Comment:
Cloning doesn't help here to only get the `public` properties, because it's still the same object scope, which means you can still access `protected`/`private` properties.

See: https://3v4l.org/p9FOu

Reply:
@jdreesen 👍 thanks for the hint, I was hoping this is a quick win 😉

Reply:
@jdreesen should be fixed now -> I tried to avoid using reflection for performance reasons, don't know if this actually matters, but just to explain my approach 😉